packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -563,6 +566,45 @@ export class TelemetryUTLogger implements ITelemetryLogger {
 }
 
 /**
+ * Take in a event object, stringify any fields that are non-primitives, and return the new event object.
+ * @param event - Event with fields you want to stringify.
+ */
+function convertToBaseEvent(event: TelemetryEventTypes): ITelemetryBaseEvent {
+    const newEvent: ITelemetryBaseEvent = {
+        category: "",
+        eventName: "",
+    };
+    for (const key of Object.keys(event)) {
+        const filteredEventVal = convertToBasePropertyType(event[key]);
+        if (filteredEventVal !== null) {
+            newEvent[key] = filteredEventVal;
+        }
+    }
+    return newEvent;
+}
+
+/**
+ * Takes in parameter, if parameter is of primitive type, return the original value.
+ * If parameter is an array, stringify then return the result.
+ * @param x - parameter passed to validate/filter
+ */
+function convertToBasePropertyType(x: any): TelemetryEventPropertyType | null {
+    switch (typeof x) {
+        case "string":
+        case "number":
+        case "boolean":
+        case "undefined":
+            return x;
+        default:
+            if (!Array.isArray(x)) {
+                return null;
+            }
+            if (x.every((val) => typeof val === "boolean" || typeof val === "string" || typeof val === "number")) {
+                return JSON.stringify(x);
+            }
+            return null;
```

Comment:
Not sure I see a value in using null here. Just return undefined and make convertToBaseEvent() not care about it - as any other undefine, it will be skipped when higher level will do strigification. If it's a problem (let's say we want to remove all undefined), then we should do it consistently across the board, i.e. including case on line 597.